proxy/coco/src/state.rs

```diff
@@ -103,40 +97,54 @@ where
     P: TryInto<PersonPayload> + Send,
     Error: From<P::Error>,
 {
-    match peer
-        .using_storage(move |store| local::default(store))
+    if let Some(owner) = default_owner(peer).await? {
+        return Ok(owner);
+    }
+
+    let payload = payload.try_into()?;
+    let pk = keys::PublicKey::from(peer.signer().public_key());
+    let delegations = Some(pk).into_iter().collect();
+    let person = peer
+        .using_storage(move |store| person::create(store, payload, delegations))
+        .await??;
+
+    let urn = person.urn();
+    let owner = peer
+        .using_storage(move |store| local::load(store, urn))
         .await??
-    {
-        Some(owner) => Ok(owner),
-        None => {
-            let pk = keys::PublicKey::from(peer.signer().public_key());
-            let payload = payload.try_into()?;
-            let person = peer
-                .using_storage(move |store| {
-                    person::create(store, payload, Some(pk).into_iter().collect())
-                })
-                .await??;
-
-            let urn = person.urn();
-            let owner = peer
-                .using_storage(move |store| local::load(store, urn))
-                .await??
-                .ok_or_else(|| Error::PersonNotFound(person.urn()))?;
-
-            {
-                let owner = owner.clone();
-                peer.using_storage(move |store| {
-                    let mut config = store.config()?;
-                    config.set_user(owner)?;
-
-                    Ok::<_, Error>(())
-                })
-                .await??;
-            }
+        .ok_or_else(|| Error::PersonNotFound(person.urn()))?;
 
-            Ok(owner)
-        },
-    }
+    set_default_owner(peer, owner.clone()).await?;
+
+    Ok(owner)
+}
+
+/// Sets a new person payload for the default owner of this [`Peer`].
+///
+/// # Errors
+///
+///   * Fails to load the default owner
+///   * Fails to verify `User`.
+///   * Fails to set the default `rad/self` for this `PeerApi`.
+#[allow(clippy::single_match_else)]
+pub async fn update_owner_payload<P>(
+    peer: &Peer<BoxedSigner>,
+    payload: P,
+) -> Result<LocalIdentity, Error>
+where
+    P: TryInto<PersonPayload> + Send,
+    Error: From<P::Error>,
+{
+    let urn = default_owner(peer).await?.ok_or(Error::MissingOwner)?.urn();
+    let payload = payload.try_into()?;
+    peer.using_storage(move |store| person::update(store, &urn, None, payload, None))
+        .await??;
+    // Update the default user info in the config with the new payload data.
```

Comment:
Let's remove the below and then this looks good to go :+1:

Reply:
Fixed